### PR TITLE
Clarify documentation regarding 'core' and 'TA' execution level

### PR DESCRIPTION
**Pull request by etienne-lms**

### CODE REVIEW

File: documentation/optee_design.md [hunk 1]
```diff
@@ -534,102 +553,110 @@ loaded TAs are preferred.
...
-Pseudo TAs and anything else directly built into OP-TEE OS do not
+Pseudo TAs and anything else directly built into OP-TEE 'core' do not
 require going through a syscall interface, since they can just link directly
-as they are directly part of the OS.
+as they are directly part of the 'core'.
+
+Most of the services defined by the "GP Core Internal API" are implemented
+through syscall from the TA to the 'core' privileged level: cryptographic
+services, communications with other TAs, ... Some services were added through
+OP-TEE development as ASCII message tracing.
```
Comment (jforissier) [inline documentation/optee_design.md]: such as

**Comment by jforissier** [comment]
There is still a typo (check the Travis log), other than that it's fine with me:
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by etienne-lms** [comment]
@jforissier: oups, i've pushed a change regarding some English contractions (something I quickly started while modifying this doc). But that out of the scope your already reviewed part! Do you prefer I remove it from this PR? (or even simply reject it?)

**Comment by jforissier** [comment]
@etienne-lms it's OK, we can continue reviewing the new commits here.

Regarding "documentation: expand some contractions", you should drop "some" and expand the single remaining one ("can't").
Also I think the commit is better justified as "Contractions are better avoided in written documentation" than with an argument about non-native speakers (for instance, this site gives the opposite point of view! https://www.vappingo.com/word-blog/when-is-it-okay-to-use-contractions-in-formal-writing/). But that's not important.

Other than that, `Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`.

**Comment by etienne-lms** [comment]
> Regarding "documentation: expand some contractions", you should drop "some" and **expand the single remaining one ("can't")**.

Hahaha! good catch :1st_place_medal: 

> Also I think the commit is better justified as "Contractions are better avoided in written documentation" 

OK. 


In the end, the goal is to ease understanding of the related informations (that sometimes is quite hard to follow). So I like this sentence from your 'vappingo' reference:
> Try to minimize the use of contractions in documents that will be read by people who speak English as a foreign language because they can confuse non-native speakers.

Let's not strictly forbid contraction, just don't (sic) use it were it's (sic) not really necessary.
Maybe my change proposal is a bit extreme.

**Comment by jforissier** [comment]
> In the end, the goal is to ease understanding of the related informations (that sometimes is quite hard to follow). So I like this sentence from your 'vappingo' reference:
>> Try to minimize the use of contractions in documents that will be read by people who speak English as a foreign language because they can confuse non-native speakers.

Doh! Indeed that's exactly your point, I'm sorry I read it wrong. Feel free to leave the commit comment as it is.

**Comment by etienne-lms** [comment]
rebased. tag applied.

File: documentation/secure_storage_rpmb.md [hunk 2]
```diff
@@ -134,7 +134,7 @@ The FAT is not encrypted.
 
 ## References
 
-- <a name="JEDECeMMC"></a>[1] _Embedded Multi-Media Card (e•MMC) Electrical Standard (5.1)_, JEDEC JESD84-B51, February 2015
+- <a name="JEDECeMMC"></a>[1] _Embedded Multi-Media Card (e-MMC) Electrical Standard (5.1)_, JEDEC JESD84-B51, February 2015
```
Comment (etienne-lms) [inline documentation/secure_storage_rpmb.md]: ok, understood. sorry i highjacked your r-b, indeed you obviously did not review it before.
i'll discard the last commit and handle it seperatly.

I though the markdown format was handy since allowing formatted doc content based on basic text which I tend to relate to ascii. You're right, it's false. Nowadays there is no reason why  plain ascii text is easier to maintain than an utf format text. i'm a bit old school, i need to upgrade my habits.

Comment (jforissier) [inline documentation/secure_storage_rpmb.md]: @etienne-lms np, thanks for the update, let's merge this now.

**Comment by etienne-lms** [comment]
thank you for your reviews, patience and frank and constructive feedbacks
especially when they highly bad practices or assumptions, wherever you're
right or wrong.
